subsys/bluetooth/controller/util/dbuf.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Generic double buffer data structure header.
+ *
+ * The structure may be used as a header for any kind of data structure that requires
+ * double buffering.
+ */
+struct dbuf_hdr {
+	/* The current element that is in use. */
+	uint8_t volatile first;
+	/* Last enqueued element. It will be used after buffer is swapped. */
+	uint8_t last;
+	/* Size in a bytes of a single element stored in double buffer. */
+	uint8_t elem_size;
+	/* Pointer for actual buffer memory. Its size should be 2 times @p elem_size. */
+	uint8_t data[0];
+};
+
+/**
+ * @brief Allocate element in double buffer.
+ *
+ * The function provides pointer to new element that may be modified by the user.
+ * For consecutive calls it always returns the same pointer until buffer is swapped
+ * by dbuf_latest_get operation.
+ *
+ * @param hdr Pointer to double buffer header.
+ * @param idx Pointer to return new element index.
+ *
+ * @return Pointer to new element memory.
+ */
+void *dbuf_alloc(struct dbuf_hdr *hdr, uint8_t *idx);
+
+/**
+ * @brief Provides pointer to last allocated element.
+ *
+ * If it is called before dbuf_alloc it will return pointer to memory that was recently
+ * enqueued.
+ *
+ * @param hdr Pointer to double buffer header.
+ *
+ * @return Pointer to last allocated element.
+ */
+static inline void *dbuf_peek(struct dbuf_hdr *hdr)
+{
+	return &hdr->data[hdr->last * hdr->elem_size];
+}
+
+/**
+ * @brief  Enqueue new element for buffer swap.
+ *
+ * @param hdr Pointer to double buffer header.
+ * @param idx Intex of element to be swapped.
+ */
+static inline void dbuf_enqueue(struct dbuf_hdr *hdr, uint8_t idx)
+{
+	hdr->last = idx;
+}
+
+/**
+ * @brief Get latest element in buffer.
+ *
+ * Latest enqueued element is pointed by last member of dbuf_hdr.
+ * If it points to a different index than member first, then buffer will be
+ * swapped and @p is_modified will be set to true.
+ *
+ * Pointer to lates element is returned.
+ *
+ * @param hdr Pointer to double buffer header.
+ * @param[out] is_modifled Pointer to return information if buffer was swapped.
```

Comment:
```suggestion
 * @param[in]  hdr Pointer to double buffer header.
 * @param[out] is_modifled Pointer to return information if buffer was swapped.
```

Reply:
Isn't `[in]` the default?

Reply:
`[in]` is usually omitted when all arguments are `in`, but when adding an `[out]` we usually add `[in]` for all `in` arguments as well. Doesn't seem to be a requirement, as we do also have `[out]`s without `[in]`, but it is more typical to add `[in]` as far as I can tell.

Reply:
I'll add `[in]` and please lets stick to one convention.